src/Dapr.SecretStore.ConfigurationExtension/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.DaprSecretStore
+{
+    /// <summary>
+    /// A Dapr Secret Store based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal class DaprSecretStoreConfigurationProvider : ConfigurationProvider
+    {
+        private const string DaprDefaultEndpoint = "127.0.0.1";
+
+        public const string SecretPath = "/v1.0/secrets";
+
+        private readonly string _store;
+
+        private readonly DaprSecretDescriptor[] _secrets;
+
+        private readonly HttpClient _httpClient;
+
+        private static string DefaultHttpPort => Environment.GetEnvironmentVariable("DAPR_HTTP_PORT") ?? "3500";
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="secrets">The secrets to retrieve.</param>
+        /// <param name="httpClient">Http client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(string store, DaprSecretDescriptor[] secrets, HttpClient httpClient)
+        {
+            if (store == null)
+            {
+                throw new ArgumentNullException(nameof(store));
+            }
+
+            if (secrets == null)
+            {
+                throw new ArgumentNullException(nameof(secrets));
+            }
+
+            if (httpClient == null)
+            {
+                throw new ArgumentNullException(nameof(httpClient));
+            }
+
+            _store = store;
+            _secrets = secrets;
+            _httpClient = httpClient;
+        }
+
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync()
+        {
+
+            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var secret in _secrets)
+            {
+                var metadata = string.Empty;
+                if (secret.Metadata != null && secret.Metadata.Count > 0)
+                {
+                    foreach (var kv in secret.Metadata)
+                    {
+                        metadata += $"{kv.Key}={kv.Value}&";
+                    }
+                    metadata = $"?{metadata.TrimEnd('&')}";
+                }
+
+                var response = await _httpClient.GetAsync($"http://{DaprDefaultEndpoint}:{DefaultHttpPort}/{SecretPath}/{_store}/{secret.SecretName}{metadata}");
```

Comment:
Dapr sdk is being refactored to make calls over gRPC and will add api to get secretes as well, The code should sue refactored client. It would be available in 0.5.0 release.

Reply:
So we'll refactor this after the 0.5.0 relase?

Reply:
Depends on #242

Reply:
Depends on #244

Reply:
Solved in 17b7966